File: archive/r/ruby/base64-encode-decode.rb
# frozen_string_literal: true

require "base64"

USAGE = "Usage: please provide a mode and a string to encode/decode"

mode, input = ARGV

abort(USAGE) if mode.nil? || input.nil? || input.strip.empty?

case mode
when "encode"
  puts Base64.strict_encode64(input)

when "decode"
  begin
    puts Base64.strict_decode64(input)
  rescue
    abort(USAGE)
  end

else
  abort(USAGE)
end

File: archive/e/elixir/base64-encode-decode.ex
case System.argv() do
  ["encode", str] when byte_size(str) > 0 ->
    Base.encode64(str) |> IO.puts()

  ["decode", str] when byte_size(str) > 0 ->
    case Base.decode64(str) do
      {:ok, res} -> IO.puts(res)
      _ -> IO.puts("Usage: please provide a mode and a string to encode/decode")
    end

  _ ->
    IO.puts("Usage: please provide a mode and a string to encode/decode")
end
